src/chunk_constraint.c

```diff
@@ -574,54 +580,70 @@ ts_chunk_constraints_add_dimension_constraints(ChunkConstraints *ccs, int32 chun
 	return cube->num_slices;
 }
 
-int
-ts_chunk_constraints_add_inheritable_constraints(ChunkConstraints *ccs, int32 chunk_id,
-												 Oid hypertable_oid)
+typedef struct ConstraintContext
 {
-	ScanKeyData skey;
-	Relation rel;
-	SysScanDesc scan;
-	HeapTuple htup;
-	int num_added = 0;
-
-	ScanKeyInit(&skey, Anum_pg_constraint_conrelid, BTEqualStrategyNumber, F_OIDEQ, hypertable_oid);
+	int num_added;
+	char chunk_relkind;
+	ChunkConstraints *ccs;
+	int32 chunk_id;
+} ConstraintContext;
 
-	rel = table_open(ConstraintRelationId, AccessShareLock);
-	scan = systable_beginscan(rel, ConstraintRelidTypidNameIndexId, true, NULL, 1, &skey);
+static ConstraintProcessStatus
+chunk_constraint_add(HeapTuple constraint_tuple, void *arg)
+{
+	ConstraintContext *cc = arg;
+	Form_pg_constraint constraint = (Form_pg_constraint) GETSTRUCT(constraint_tuple);
 
-	while (HeapTupleIsValid(htup = systable_getnext(scan)))
+	if (chunk_constraint_need_on_chunk(cc->chunk_relkind, constraint))
 	{
-		Form_pg_constraint pg_constraint = (Form_pg_constraint) GETSTRUCT(htup);
-
-		if (chunk_constraint_need_on_chunk(pg_constraint))
-		{
-			chunk_constraints_add(ccs, chunk_id, 0, NULL, NameStr(pg_constraint->conname));
-			num_added++;
-		}
+		chunk_constraints_add(cc->ccs, cc->chunk_id, 0, NULL, NameStr(constraint->conname));
+		return CONSTR_PROCESSED;
 	}
 
-	systable_endscan(scan);
-	table_close(rel, AccessShareLock);
+	return CONSTR_IGNORED;
+}
+
+TSDLLEXPORT int
```

Comment:
Using `TSDLLEXPORT` in source breaks discovering definitions in VS Code and it is usually avoided.
```suggestion
int
```